plugins/spu2-x/src/Windows/SndOut_XAudio2.inl

```diff
@@ -182,7 +181,7 @@ private:
             }
 
             EnterCriticalSection(&cs);
-            safe_delete_array(qbuffer);
+            m_buffer = nullptr;
```

Comment:
I don't think the buffer is used by any other thread when this code is reached. That's what the documentation suggested to me but I didn't remove the code just in case I'm wrong.

I didn't change the critical section to a std::mutex since I was wondering whether it was necessary in the first place (FWIW I looked at Dolphin's XAudio2 code and they don't use any mutexes).